### PR TITLE
Ben/rendering

### PR DESCRIPTION
**Pull request by enochtsang**

### CODE REVIEW

File: src/AssetManager.cpp [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+#include "assimp/Importer.hpp"
+#include "assimp/scene.h"
+#include "assimp/postprocess.h"
+
+#include "glm/gtc/type_ptr.hpp"
+
+#include <iostream>
+
+AssetManager::AssetManager() {
```
Comment (ammunoz) [inline src/AssetManager.cpp]: Is there any benefit to the empty constructor over the implicit?

File: src/Renderer.cpp [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+    //time += 0.01f;
+
+    //This should be moved to a different system
+    SDL_Event event;
+
+    while (SDL_PollEvent(&event)) {
+        switch (event.type) {
+            case SDL_QUIT:
+                std::cout << "SDL_QUIT was called" << std::endl;
```
Comment (ammunoz) [inline src/Renderer.cpp]: I would prefer that all quit logic be in one place. What I've done so far is defined a quit() function for the different systems and called those in the main loop. We can move all of them out altogether later on.

File: src/Renderer.cpp [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+void Renderer::handle_key_press(const Event& e) {
+    int player_id = e.get_value<int>("player_id", -1);
+    int key = e.get_value<int>("key", -1);
+    int value = e.get_value<int>("value", 0);
+
+    glm::mat4 transform;
+
+    switch (key) {
+        case SDLK_LEFT:
```
Comment (ammunoz) [inline src/Renderer.cpp]: Just a warning: same keys are mapped for  audio testing. Might be a little annoying when we're testing the camera

File: src/Renderer.cpp [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+
+    endRender();
+}
+
+bool Renderer::init_window() {
+    SDL_Init(SDL_INIT_EVERYTHING);
+
+    int sdl_flags = SDL_WINDOW_SHOWN | SDL_WINDOW_OPENGL;
+
+    int screen_width = 640;
```
Comment (ammunoz) [inline src/Renderer.cpp]: Should be const

File: src/Renderer.h [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+#include "EventSystem.h"
+#include "RenderingComponent.h"
+#include "Shader.h"
+
+#include "SDL_opengl.h"
+#include "SDL.h"
+
+class AssetManager;
+
+class Renderer : public EventSystem<Renderer> {
```
Comment (ammunoz) [inline src/Renderer.h]: Should this not be called RenderingSystem?

File: src/AssetManager.cpp (hunk 1 above)
Comment (enochtsang) [reply src/AssetManager.cpp]: I'd agree with having an explicit one, i think it's easier to read even if it's just a little bit.

**Pull request merged**